src/main/kotlin/io/github/yuutoproject/yuutobot/commands/minigame/MinigameInstance.kt

```diff
@@ -0,0 +1,182 @@
+
+        begun = true
+        progress(event)
+    }
+
+    fun progress(event: GuildMessageReceivedEvent) {
+        if (rounds > maxRounds) {
+            endGame(event)
+            return
+        }
+
+        // Unfortunately, no removeOrNull, so we have to use a try/catch
+        try {
+            currentQuestion = questions.removeAt(0)
+        } catch (e: IndexOutOfBoundsException) {
+            endGame(event)
+            return
+        }
+
+        currentAnswers = currentQuestion.answers.map { it.toLowerCase() }
+            .toMutableList()
+
+        if (currentQuestion.type == "FILL") {
+            event.channel.sendMessage(currentQuestion.question).queue()
+        } else if (currentQuestion.type == "MULTIPLE") {
+            val questionString = "${currentQuestion.question}\n"
+
+            val answerString = (currentQuestion.wrong + currentQuestion.answers).shuffled()
+                .mapIndexed { i, answer ->
+                    if (currentAnswers.contains(answer.toLowerCase())) {
+                        currentAnswers.add((i + 1).toString())
+                    }
+
+                    "${i + 1}) $answer"
+                }.joinToString("\n")
+
+            event.channel.sendMessage(questionString + answerString).queue()
+        }
+    }
+
+    private fun endGame(event: GuildMessageReceivedEvent) {
+        // Sort by descending value and then map each value to a line in the scoreboard, then join it
+        val scoreboard = players.entries.sortedByDescending { it.value }.mapIndexed { i, entry ->
+            "${i + 1}) <@${entry.key}> with ${entry.value} points"
+        }.joinToString("\n")
+
+        val embed = EmbedBuilder()
+            .setColor(Color.BLUE)
+            .setTitle("Minigame ended!")
+            .setDescription("Total points:\n$scoreboard")
+        event.channel.sendMessage(embed.build()).queue()
+
+        manager.unregister(this)
+    }
+
+    fun answerReceived(event: GuildMessageReceivedEvent) {
+        // A new guess is made, so we reset the stale-game timer
+        timer = System.currentTimeMillis()
+
+        if (currentAnswers.contains(event.message.contentStripped.toLowerCase())) {
+            players[event.author.idLong] = players[event.author.idLong]!! + 1
+            event.channel.sendMessage("${event.author.name} got the point!").queue()
+            rounds += 1
+            progress(event)
+        }
+    }
+
+    fun reactionRecv(event: GuildMessageReactionAddEvent) {
+        if (
+            event.user.isBot ||
+            players.contains(event.user.idLong) ||
+            begun ||
+            event.messageId != startingMessageID
+        ) return
+
+        players[event.user.idLong] = 0
+    }
+
+    fun reactionRetr(event: GuildMessageReactionRemoveEvent) {
+        if (
+            event.messageId == startingMessageID &&
+            !begun
+        ) {
+            players.remove(event.user!!.idLong)
+        }
+    }
```

Comment:
The bot does not check what reaction you add/take away, removing another (different/not the intended) reaction will also invalidate your join and you can join with any reaction you want. 
This bug could be a feature, but it would be a weird one in my opinion.

*PS: I'm 100% sure if this is the part of the code that needs changing*

Reply:
This error is only half fixed in the current latest (fb43211) commit. Only the removing checks for the correct emote.